### PR TITLE
add charity model

### PR DESCRIPTION
**Pull request by ajin12**

### CODE REVIEW

File: app/src/main/java/com/example/una/models/Charity.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        Charity charity = new Charity();
+
+        // extract the values from JSON
+        charity.ein = jsonObject.getString("ein");
+        charity.name = jsonObject.getString("charityName");
+        charity.description = jsonObject.getString("tagLine");
+        charity.mission = jsonObject.getString("mission");
+        charity.category = jsonObject.getString("categoryName");
+        charity.cause = jsonObject.getString("causeName");
+        charity.donationAddress = (Address) jsonObject.get("donationAddress");
```
Comment (chiped) [inline app/src/main/java/com/example/una/models/Charity.java]: do you have a source for your model?

I don't think their `donationAddress` will be of type `android.location.Address`

File: app/src/main/java/com/example/una/models/Charity.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    // attributes
+    public String ein; // federal Employer Identification Number
+    public String name;
+    public String description; // tag line describing the charity
+    public String mission; // organization's mission statement
+    public String category;
+    public String cause;
+    public Address donationAddress;
+    public boolean isFeatured;
+    // TODO - add donorList, photoUrl, verified
```
Comment (chiped) [inline app/src/main/java/com/example/una/models/Charity.java]: create an issue to keep track of this todo

File: app/src/main/java/com/example/una/models/Charity.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        charity.name = jsonObject.getString("charityName");
+        charity.description = jsonObject.getString("tagLine");
+        charity.mission = jsonObject.getString("mission");
+        charity.category = jsonObject.getString("categoryName");
+        charity.cause = jsonObject.getString("causeName");
+        charity.isFeatured = false;
+
+        // set charity location
+        charity.donationAddress = new Address(Locale.ENGLISH);
+        charity.donationAddress.setAddressLine(0, jsonObject.getString("streetAddress1"));
```
Comment (chiped) [inline app/src/main/java/com/example/una/models/Charity.java]: I don't think this is accurate.

Looking at the doc, seems like they have two addresses `mailingAddress` and `donationAddress`.
Each of that field has the details. Calling `jsonObject.getString("streetAddress1")` on the outer object of Charity will return `null`

**Comment by chiped** [comment]
have you tried making this request to see what data it returns? You can register your app as mentioned on https://charity.3scale.net/docs/data-api/reference#working-with-the-api and make the request in a browser to see how the returned data is structured

File: app/src/main/java/com/example/una/models/Charity.java (hunk 3 above)
Comment (ajin12) [reply app/src/main/java/com/example/una/models/Charity.java]: Would `charity.donationAddress.setAddressLine(0, jsonObject.getJSONObject("donationAddress").getString("streetAddress1"));` for line 38 be correct?

Comment (chiped) [reply app/src/main/java/com/example/una/models/Charity.java]: Yes, if donation address is what you want to store here. You might want to consider the case when they don't have a `donationAddress` set in the server

**Comment by ajin12** [comment]
I just realized I was incorrectly getting categoryName and causeName as well, so I changed accordingly

**Comment by chiped** [comment]
Don't forget to **squash and merge**

**Pull request merged**